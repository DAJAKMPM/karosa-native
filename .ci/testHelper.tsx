import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
