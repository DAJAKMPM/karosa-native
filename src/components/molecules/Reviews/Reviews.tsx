/**
 *
 * Reviews
 * @format
 *
 */

import React, { FC } from "react";
import { View } from "react-native";
import Text from "@app/atoms/Text";
import Rating from "@app/atoms/Rating";
import Image from "@app/atoms/Image";
import ExpandableText from "@app/molecules/ExpandableText";
import ImageOverlayReviews from "@app/organisms/ImageOverlayReviews";

import type { PropsType } from "./types";
import ReviewsStyles from "./styles";

const Reviews: FC<PropsType> = (props) => {
  const {
    avatarPhoto,
    date,
    feedback,
    rate,
    variation,
    overlay,
    username,
    hasBottomDivider,
    ratingImage,
    ratingType,
  } = props;

  return (
    <View
      style={[
        ReviewsStyles.container,
        hasBottomDivider ? ReviewsStyles.divider : null,
      ]}
    >
      <View style={ReviewsStyles.column}>
        <Image
          source={{
            uri: avatarPhoto,
          }}
          imageStyle={ReviewsStyles.avatar}
          resizeMode="cover"
        />
      </View>
      <View style={ReviewsStyles.details}>
        <Text text={username} textStyle={ReviewsStyles.txtUsername} />

        <View style={ReviewsStyles.primaryMargin}>
          <Rating
            ratingCount={5}
            readonly
            imageSize={10}
            startingValue={rate}
            ratingImage={ratingImage}
            type={ratingType}
          />
        </View>

        <View style={ReviewsStyles.primaryMargin}>
          <ExpandableText
            text={feedback}
            readMoreStyle={ReviewsStyles.readMore}
          />
        </View>

        {overlay && (
          <View style={ReviewsStyles.secondaryMargin}>
            <ImageOverlayReviews overlayProps={overlay} />
          </View>
        )}

        <Text
          text={`${date}, Variation: ${variation}`}
          textStyle={ReviewsStyles.txtVariation}
        />
      </View>
    </View>
  );
};

export default Reviews;
