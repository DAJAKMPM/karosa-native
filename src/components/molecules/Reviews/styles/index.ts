/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme } from "@app/styles";

const ReviewsStyles = StyleSheet.create({
  varietyImage: {
    width: 70,
    height: 70,
    marginRight: 10,
  },
  sellerImage: {
    height: 50,
    width: 50,
    borderRadius: 30,
  },

  txtBlackRegular: {
    ...theme.textRegular,
  },
  txtReviewParagraph: {
    ...theme.textRegular,
    marginHorizontal: 16,
  },
  tagsStyle: {
    backgroundColor: theme.colors.dark5,
    margin: 3,
    padding: 8,
    borderRadius: 16,
    ...theme.textLight,
    color: theme.colors.dark15,
  },

  subContainer: {
    backgroundColor: theme.colors.white,
    flex: 1,
    marginBottom: 10,
  },

  reviewContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    margin: 8,
  },
  txtNameAddress: {
    flexDirection: "column",
    justifyContent: "center",
    marginLeft: 8,
  },

  iconStyle: {
    width: "100%",
    height: 30,
  },
  buyerContainer: {
    flexDirection: "row",
  },
  likedIcon: {
    marginLeft: 10,
  },
  tagsContainer: {
    flexDirection: "row",
    flexWrap: "wrap",
    margin: 8,
  },
  buyerGallery: {
    flexDirection: "row",
    margin: 8,
  },
  commentContainer: {
    marginHorizontal: 25,
  },
  ratingContainer: {
    width: 80,
  },
});

export default ReviewsStyles;
