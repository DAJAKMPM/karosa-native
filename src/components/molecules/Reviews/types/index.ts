/**
 *
 * @format
 *
 */

import { ReactNode } from "react";
import { OverlayPropsType } from "@app/organisms/ImageOverlayReviews/types";

export type PropsType = {
  avatarPhoto: string;
  username: string;
  rate: number;
  feedback: string;
  date: string;
  variation: string;
  overlay?: OverlayPropsType[];
  hasBottomDivider?: boolean;
  ratingImage?: ReactNode;
  ratingType?: string;
};
