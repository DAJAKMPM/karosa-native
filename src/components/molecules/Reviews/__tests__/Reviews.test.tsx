import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import Reviews from "../Reviews";

const FAKE_URI = "https://i.imgur.com/0y8Ftya.jpg";

describe("Reviews Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <Reviews
        avatarPhoto={FAKE_URI}
        date="2022-12-12"
        feedback="test"
        rate={2}
        ratingType="custom"
        username="testUsername"
        variation="2"
        ratingImage={{
          source: {
            uri: FAKE_URI,
          },
        }}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
