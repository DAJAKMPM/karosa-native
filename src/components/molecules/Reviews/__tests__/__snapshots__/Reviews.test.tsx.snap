// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "flexDirection": "row",
          "paddingLeft": 14,
          "paddingVertical": 18,
        },
        null,
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <Image
          onLoad={[Function]}
          resizeMode="cover"
          source={
            Object {
              "uri": "https://i.imgur.com/0y8Ftya.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 100,
              "bottom": 0,
              "height": 50,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 50,
            }
          }
          testID="RNE__Image"
          theme={
            Object {
              "colors": Object {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": Object {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
          transition={true}
          transitionDuration={360}
        />
        <View
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#bdbdbd",
                "borderRadius": 100,
                "height": 50,
                "justifyContent": "center",
                "width": 50,
              }
            }
            testID="RNE__Image__placeholder"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <ActivityIndicator
                animating={true}
                color="#1C1C1C"
                hidesWhenStopped={true}
                size="small"
              />
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 50,
                "width": 50,
              },
              Object {
                "borderRadius": 100,
                "height": 50,
                "width": 50,
              },
            ]
          }
          testID="RNE__Image__children__container"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "marginLeft": 14,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#4F4F4F",
            "fontFamily": "SFProText-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 18.75,
            "marginBottom": 14,
          }
        }
      >
        testUsername
      </Text>
      <View
        style={
          Object {
            "marginBottom": 14,
          }
        }
      >
        <View
          pointerEvents="none"
        >
          <View
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              onLayout={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#f1c40f",
                    "height": 10,
                    "width": 20,
                  }
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "height": 10,
                    "width": 30,
                  }
                }
              />
            </View>
            <View>
              <Image
                source={
                  Object {
                    "source": Object {
                      "uri": "https://i.imgur.com/0y8Ftya.jpg",
                    },
                  }
                }
                style={
                  Object {
                    "height": 10,
                    "tintColor": undefined,
                    "width": 10,
                  }
                }
              />
            </View>
            <View>
              <Image
                source={
                  Object {
                    "source": Object {
                      "uri": "https://i.imgur.com/0y8Ftya.jpg",
                    },
                  }
                }
                style={
                  Object {
                    "height": 10,
                    "tintColor": undefined,
                    "width": 10,
                  }
                }
              />
            </View>
            <View>
              <Image
                source={
                  Object {
                    "source": Object {
                      "uri": "https://i.imgur.com/0y8Ftya.jpg",
                    },
                  }
                }
                style={
                  Object {
                    "height": 10,
                    "tintColor": undefined,
                    "width": 10,
                  }
                }
              />
            </View>
            <View>
              <Image
                source={
                  Object {
                    "source": Object {
                      "uri": "https://i.imgur.com/0y8Ftya.jpg",
                    },
                  }
                }
                style={
                  Object {
                    "height": 10,
                    "tintColor": undefined,
                    "width": 10,
                  }
                }
              />
            </View>
            <View>
              <Image
                source={
                  Object {
                    "source": Object {
                      "uri": "https://i.imgur.com/0y8Ftya.jpg",
                    },
                  }
                }
                style={
                  Object {
                    "height": 10,
                    "tintColor": undefined,
                    "width": 10,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 14,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={4}
          onTextLayout={[Function]}
          style={
            Object {
              "color": "#1C1C1C",
              "fontFamily": "SFProText-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 18.75,
            }
          }
        >
          test
        </Text>
        <Text
          onPress={[Function]}
          style={
            Object {
              "color": "#4F4F4F",
              "fontFamily": "SFProText-Regular",
              "fontSize": 12,
              "fontWeight": "300",
              "lineHeight": 14.06,
            }
          }
        >
          Read More
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#4F4F4F",
            "fontFamily": "SFProText-Regular",
            "fontSize": 12,
            "fontWeight": "400",
            "lineHeight": 14.06,
          }
        }
      >
        2022-12-12, Variation: 2
      </Text>
    </View>
  </View>
</View>
`;
